TextAnalyzer/Main.cs

```diff
@@ -22,37 +22,53 @@ public Main()
             optionsList.Add(new CountPunctuationMarksCommand());
             optionsList.Add(new CountEveryLetterReportCommand());
             optionsList.Add(new CountSentencesInFileCommand());
-			      optionsList.Add(new CreateReportCommand());
+            optionsList.Add(new CreateReportCommand());
             optionsList.Add(new ExitCommand());
           
-		        StartMenu();
+		    StartMenu();
         }
 
         public void StartMenu()
         {
-            if (TextFileFetcher.FILE_NAME == null)
+            if (TextFileFetcher.CheckIfFileExists())
             {
-                writeMessege("No File Loaded");
+                writeMessege("Opened File:" + TextFileFetcher.FileName);
+                WriteOptionsAndTakeChoice(optionsList);
             }
             else
             {
-                writeMessege("Opened File:" + TextFileFetcher.FILE_NAME);
+                writeMessege("No File Loaded");
+                WriteOptionsAndTakeChoice(GetOnFileNotDownloadedOptions());
             }
+            
+
+        }
+
+        private List<ICommand> GetOnFileNotDownloadedOptions()
+        {
+            var downloadIndex = optionsList.FindIndex((e) => e.GetType() == typeof(DownloadFileCommand));
+            var exitIndex = optionsList.FindIndex((e) => e.GetType() == typeof(ExitCommand));
 
+            List<ICommand> options = new List<ICommand>{ optionsList[downloadIndex], optionsList[exitIndex] };
+            return options;
+        }
+
+        private void WriteOptionsAndTakeChoice(List<ICommand> viableOptions)
+        {
             int i = 1;
-            foreach (var command in optionsList) writeMessege($"{i++}. {command.Description}");
+            foreach (var command in viableOptions) writeMessege($"{i++}. {command.Description}");
```

Comment:
I suggest to use more strict variables and more readable form
```suggestion
            foreach (ICommandin viableOptions)
            {
                    writeMessege($"{i++}. {command.Description}"
            );
```

Reply:
I agree, especially about the form, though it was switched from a this type of convention to the one used before in one of the early PR as part of the review.
Also, i think you are missing an identifier and a space in this foreach loop, so I suggest something like this:
```suggestion
            foreach (ICommand command in viableOptions)
            {
                    writeMessege($"{i++}. {command.Description}"
            );
```

Reply:
It also seems that there was a missing semicolon, we have to be carefull with suggstions via github as they bring potential risks.